packages/cli/lib/localDev.js

```diff
@@ -0,0 +1,376 @@
+      text: i18n(`${i18nKey}.status.creatingProject`, {
+        accountIdentifier: uiAccountDescription(targetAccountId),
+        projectName: projectConfig.name,
+      }),
+    });
+
+    try {
+      await createProject(targetAccountId, projectConfig.name);
+      SpinniesManager.succeed('createProject', {
+        text: i18n(`${i18nKey}.status.createdProject`, {
+          accountIdentifier: uiAccountDescription(targetAccountId),
+          projectName: projectConfig.name,
+        }),
+        succeedColor: 'white',
+      });
+    } catch (err) {
+      SpinniesManager.fail('createProject');
+      logger.log(i18n(`${i18nKey}.status.failedToCreateProject`));
+      process.exit(EXIT_CODES.ERROR);
+    }
+  } else {
+    // We cannot continue if the project does not exist in the target account
+    logger.log();
+    logger.log(i18n(`${i18nKey}.logs.choseNotToCreateProject`));
+    process.exit(EXIT_CODES.SUCCESS);
+  }
+};
+
+// Create an initial build if the project was newly created in the account
+// Return the newly deployed build
+const createInitialBuildForNewProject = async (
+  projectConfig,
+  projectDir,
+  targetAccountId
+) => {
+  const initialUploadResult = await handleProjectUpload(
+    targetAccountId,
+    projectConfig,
+    projectDir,
+    (...args) => pollProjectBuildAndDeploy(...args, true),
+    i18n(`${i18nKey}.logs.initialUploadMessage`)
+  );
+
+  if (initialUploadResult.uploadError) {
+    if (
+      isSpecifiedError(initialUploadResult.uploadError, {
+        subCategory: PROJECT_ERROR_TYPES.PROJECT_LOCKED,
+      })
+    ) {
+      logger.log();
+      logger.error(i18n(`${i18nKey}.errors.projectLockedError`));
+      logger.log();
+    } else {
+      logApiErrorInstance(
+        initialUploadResult.uploadError,
+        new ApiErrorContext({
+          accountId: targetAccountId,
+          projectName: projectConfig.name,
+        })
+      );
+    }
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  if (!initialUploadResult.succeeded) {
+    let subTasks = [];
+
+    if (initialUploadResult.buildResult.status === 'FAILURE') {
+      subTasks =
+        initialUploadResult.buildResult[PROJECT_BUILD_TEXT.SUBTASK_KEY];
+    } else if (initialUploadResult.deployResult.status === 'FAILURE') {
+      subTasks =
+        initialUploadResult.deployResult[PROJECT_DEPLOY_TEXT.SUBTASK_KEY];
+    }
+
+    const failedSubTasks = subTasks.filter(task => task.status === 'FAILURE');
+
+    logger.log();
+    failedSubTasks.forEach(failedSubTask => {
+      console.error(failedSubTask.errorMessage);
+    });
+    logger.log();
+
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  return initialUploadResult.buildResult;
+};
+
+module.exports = {
+  confirmDefaultAccountIsTarget,
+  checkCorrectParentAccountType,
+  suggestRecommendedNestedAccount,
+  createSandboxForLocalDev,
+  createDeveloperTestAccountForLocalDev,
+  createNewProjectForLocalDev,
+  createInitialBuildForNewProject,
+};
```

Comment:
The logic in this file is pretty close to 1:1 with the old version of `dev.js`. With the exception of new logic for public apps in the account selection functions, nothing was changed, just moved